internal/core/src/expr/ITypeExpr.h

```diff
@@ -235,6 +318,23 @@ class UnaryRangeFilterExpr : public ITypeFilterExpr {
         return ss.str();
     }
 
+    void
+    GatherInfo(ExprInfo& info) const override {
+        if (IsMaterializedViewSupported(column_.data_type_)) {
+            info.field_id_to_values[column_.field_id_.get()].insert(val_);
+
+            if (op_type_ != proto::plan::OpType::Equal) {
+                if (op_type_ == proto::plan::OpType::NotEqual) {
+                    info.has_not = true;
+                } else {
+                    // For unary range filter <, >, <=, >=
+                    // we add a dummy value to indicate multiple values
+                    info.field_id_to_values[column_.field_id_.get()].emplace();
```

Comment:
if (op_type_ == proto::plan::OpType::NotEqual) {
         info.has_not = true;
   }
   if (op_type_ == proto::plan::OpType::Equal ||
        op_type_ == proto::plan::OpType:NotEqual) {
        info.field_id_to_values[column_.field_id_.get()].insert(val_);
     } else {
         info.field_id_to_values[column_.field_id_.get()].emplace()
     }
upper code is more clear and avoid double insert

Reply:
Double insertion is intentional. I will add more comments.

Reply:
good